# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

## My Lane

I chose the Content Refresh Priority Prediction lane.

The objective of this project is to identify website pages that should be refreshed before their search performance declines. This project combines machine learning with SEO analytics to help prioritize content updates based on historical search data.



## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## Research Question

### Decision
Which website pages should be refreshed first to reduce future traffic loss?

### Action
The SEO or content team will update the pages recommended by the model.

### Cost of a Wrong Recommendation
If the model recommends the wrong pages, time and effort may be wasted on unnecessary updates. If it fails to identify important declining pages, website traffic and user engagement may decrease.

### Why Machine Learning?
Machine learning can analyze multiple factors such as impressions, CTR, average position, and content age together to identify patterns that are difficult to detect manually.



## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [9]:
import os
print(os.getcwd())

/content/flyrank-ml-internship


In [10]:
!git clone https://github.com/ganeshmpsmg/flyrank-ml-internship.git

Cloning into 'flyrank-ml-internship'...
remote: Enumerating objects: 100, done.
remote: Counting objects: 100% (100/100), done.
remote: Compressing objects: 100% (71/71), done.
remote: Total 100 (delta 20), reused 80 (delta 13), pack-reused 0 (from 0)
Receiving objects: 100% (100/100), 1.82 MiB | 15.12 MiB/s, done.
Resolving deltas: 100% (20/20), done.


In [11]:
%cd flyrank-ml-internship

/content/flyrank-ml-internship/flyrank-ml-internship


In [12]:
import os
print(os.getcwd())

/content/flyrank-ml-internship/flyrank-ml-internship


In [13]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Rows:", df.shape[0])
print("Columns:", df.shape[1])

print("\nTrend Distribution")
print(df["trend_direction"].value_counts())

print("\nCorrelation")
print(df["search_volume"].corr(df["impressions_90d"]))

Rows: 30000
Columns: 44

Trend Distribution
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Correlation
0.0012030310696208905


## Quick Look at the Data

The starter dataset contains **30,000 pages** with **44 features**, providing a good foundation for machine learning analysis.

The trend distribution shows that **16,262 pages are marked as "down"**, while **5,962 are "stable"**, **4,388 are "up"**, **2,236 are "new"**, and **1,152 are "flat"**. This indicates that many pages experience declining performance, making content refresh prioritization an important problem.

The correlation between **search volume** and **impressions** is approximately **0.0012**, which is very close to zero. This suggests that search volume alone is not enough to estimate page performance, so combining multiple SEO features using machine learning could provide better decision support.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project provides decision support by identifying patterns in historical SEO data.

The model can estimate which pages may benefit from a content refresh based on observed data.

The project cannot prove why Google ranks pages in a certain way or guarantee future rankings. The recommendations should assist human decision-making rather than replace it.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.